lib/bump/bump.ts

```diff
@@ -0,0 +1,180 @@
+import cloudwatch = require('@aws-cdk/aws-cloudwatch');
+import cbuild = require('@aws-cdk/aws-codebuild');
+import events = require('@aws-cdk/aws-events');
+import cdk = require('@aws-cdk/cdk');
+import { createBuildEnvironment } from '../build-env';
+import permissions = require('../permissions');
+import { WritableGitHubRepo } from '../repo';
+
+export interface AutoBumpOptions {
+  /**
+   * The command to execute in order to bump the repo.
+   *
+   * The bump command is responsible to bump any version metadata, update
+   * CHANGELOG and commit this to the repository.
+   *
+   * @default "/bin/sh ./bump.sh"
+   */
+  bumpCommand?: string;
+
+  /**
+   * The schedule to produce an automatic bump.
+   *
+   * The expression can be one of:
+   *
+   *  - cron expression, such as "cron(0 12 * * ? *)" will trigger every day at 12pm UTC
+   *  - rate expression, such as "rate(1 day)" will trigger every 24 hours from the time of deployment
+   *
+   * To disable, use the string `disable`.
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html
+   * @default "cron(0 12 * * ? *)" (12pm UTC daily)
+   */
+  scheduleExpression?: string;
+
+  /**
+   * The image used for the builds.
+   *
+   * @default superchain (see docs)
+   */
+  buildImage?: cbuild.IBuildImage;
+
+  /**
+   * The type of compute to use for this build.
+   * See the {@link ComputeType} enum for the possible values.
+   *
+   * @default taken from {@link #buildImage#defaultComputeType}
+   */
+  computeType?: cbuild.ComputeType;
+
+  /**
+   * Indicates how the project builds Docker images. Specify true to enable
+   * running the Docker daemon inside a Docker container. This value must be
+   * set to true only if this build project will be used to build Docker
+   * images, and the specified build environment image is not one provided by
+   * AWS CodeBuild with Docker support. Otherwise, all associated builds that
+   * attempt to interact with the Docker daemon will fail.
+   *
+   * @default false
+   */
+  privileged?: boolean;
+
+  /**
+   * Environment variables to pass to build
+   */
+  env?: { [key: string]: string };
+
+  /**
+   * The name of the branch to push the bump commit (e.g. "master")
+   * This branch has to exist.
+   *
+   * @default the commit will be *forced* pushed to the branch `bump/$(git describe)`
+   */
+  branch?: string;
+}
+
+export interface AutoBumpProps extends AutoBumpOptions {
+  /**
+   * The repository to bump.
+   */
+  repo: WritableGitHubRepo;
+}
+
+export class AutoBump extends cdk.Construct {
+
+  /**
+   * CloudWatch alarm that will be triggered if bump fails.
+   */
+  public readonly alarm: cloudwatch.Alarm;
+
+  constructor(parent: cdk.Construct, id: string, props: AutoBumpProps) {
+    super(parent, id);
+
+    const bumpCommand = props.bumpCommand || '/bin/sh ./bump.sh';
```

Comment:
Please just `bash`? There is no reasonable system today that doesn't have `bash` and I don't want to be screwed over by having using a bashism.

Reply:
`/bin/bash` you mean?

Reply:
Yes